tests/fail/external_C/function_not_in_SO.rs

```diff
@@ -0,0 +1,13 @@
+//@only-target-linux
+//@only-on-host
+//@compile-flags: -Zmiri-external_c_so_file=tests/external_C/libtestlib.so
```

Comment:
Please make sure this test also works when Miri is run via `x.py` in the rustc workspace.
I think currently it will not work since you are doing some setup in `./miri`. The setup needs to happen in `tests/compiletest.rs` instead.